src/main/webapp/app/shared/modal/AddMutationModal.tsx

```diff
@@ -0,0 +1,649 @@
+import Tabs from 'app/components/tabs/tabs';
+import React, { KeyboardEventHandler, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { Alert, Button, Col, Modal, ModalBody, ModalFooter, Row } from 'reactstrap';
+import CreatableSelect from 'react-select/creatable';
+import ValidatedForm from 'app/shared/form/ValidatedForm';
+import { ValidatedField, ValidatedSelect } from 'app/shared/form/ValidatedField';
+import { notifyError } from 'app/oncokb-commons/components/util/NotificationUtils';
+import _ from 'lodash';
+import { generateUuid, parseAlterationName } from '../util/utils';
+import { IRootStore } from 'app/stores';
+import { componentInject } from '../util/typed-inject';
+import { flow, flowResult } from 'mobx';
+import { CancellablePromise } from 'mobx/dist/internal';
+import { EntityStatusAlteration } from '../api/generated';
+import './add-mutation-modal.scss';
+import { IGene } from '../model/gene.model';
+import { FaChevronDown, FaChevronUp, FaExclamationTriangle } from 'react-icons/fa';
+import { Alteration, Mutation } from '../model/firebase/firebase.model';
+
+// Follow naming convention: all words start with uppercase, and space are denoted by _
+// This is used to populate field names
+type MutationState = {
+  label: string;
+  value: string;
+  uuid: string;
+  Alteration: string;
+  Name: string;
+  Protein_Change: string;
+  Protein_Start: string;
+  Protein_End: string;
+  Ref_Residues: string;
+  Var_Residues: string;
+  Consequence: string;
+  Comment?: string;
+  Excluding: MutationState[];
```

Comment:
why do we upper case the first letter? What naming convention are you referring to?